src/excited_workflow/source_datasets/land_cover.py

```diff
@@ -1,19 +1,128 @@
 """Ingest Copernicus land cover data."""
+from pathlib import Path
 from typing import Literal
 
+import numpy as np
 import xarray as xr
 import xarray_regrid  # noqa: F401
 
 from excited_workflow.source_datasets.protocol import DataSource
 from excited_workflow.source_datasets.protocol import get_freq_kw
 
 
+def _cftime_to_datetime(data: xr.DataArray) -> np.ndarray:
+    """Convert cftime dataarray values to a numpy datetime format.
+
+    Args:
+        data: DataArray containing the time values, e.g. ds["time"].
+
+    Returns:
+        Numpy array with a datetime64 dtype.
+    """
+    return np.array([np.datetime64(el) for el in data.to_numpy()])
+
+
+def regrid(
+    file: Path,
+    target_grid: xr.Dataset,
+    variables: list[str] | None = None,
+) -> xr.Dataset:
+    """Regrids one file to target dataset.
+
+    Args:
+        file: file to regrid
+        variables: List of variable names which should be downloaded.
+        target_grid: Grid to which the data should be regridded to.
+
+    Returns:
+        Regridded Dataset.
+    """
+    ds = xr.open_dataset(file, chunks={"lat": 2000, "lon": 2000})
+
+    # Set time to middle of bounds.
+    time_coords = _cftime_to_datetime(ds["time_bounds"].mean(dim="bounds"))
+    ds = ds.drop("time_bounds")
+    ds["time"] = time_coords
+
+    ds = ds[["lccs_class"]]  # Only take the class variable.
+    ds = ds.sortby(["lat", "lon"])
+    ds = ds.rename({"lat": "latitude", "lon": "longitude"})
+
+    if variables is not None:
+        ds = ds[variables]
+
+    ds = ds.regrid.most_common(target_grid, time_dim="time")
+
+    return ds
+
+
+def check_dims(dat: xr.Dataset, target: xr.Dataset, var: str) -> bool:
```

Comment:
```suggestion
def coord_matches(dat: xr.Dataset, target: xr.Dataset, coord: str) -> bool:
```
On a second look, it's the coordinate (and the values) which are checked. It's probably better to reflect that in the definition & docstring